pkg/common/flyte_url.go

```diff
@@ -0,0 +1,92 @@
+package common
+
+import (
+	"fmt"
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/admin"
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/core"
+	"regexp"
+	"strconv"
+)
+
+type IOType string
```

Comment:
you can use `//go:generate enumer` and it will autogenerate the lookup as well. you wont need to 
example: https://github.com/flyteorg/flytepropeller/blob/dd49c9df317b9830c98641106d3b2cdd8fa18eb2/manager/config/config.go#L10